src/redux/sagas/motions/claimAllMotionRewards.ts

```diff
@@ -41,32 +40,36 @@ function* claimAllMotionRewards({
         colonyAddress,
       );
 
-    const {
-      data: { getActionsByColony },
-    }: ApolloQueryResult<GetColonyActionsQuery> = yield apolloClient.query<
-      GetColonyActionsQuery,
-      GetColonyActionsQueryVariables
-    >({
-      query: GetColonyActionsDocument,
-      variables: {
-        colonyAddress,
-        filter: { isMotion: { eq: true } },
-      },
-    });
-
-    const motions = getActionsByColony?.items.filter(notNull);
-
-    if (!motions) {
-      throw new Error('Could not retrieve motions from database');
-    }
-
     const databaseMotionIds = motionIds.map((nativeMotionId) =>
       getMotionDatabaseId(chainId, votingClient.address, nativeMotionId),
     );
 
+    const motions: ColonyMotion[] = [];
+
+    for (const databaseMotionId of databaseMotionIds) {
+      const {
+        data: { getColonyMotion },
+      }: ApolloQueryResult<GetColonyMotionQuery> = yield apolloClient.query<
```

Comment:
I don't think `ApolloQueryResult` is necessary here - what you put as the first generic should already provide typing for the response

Reply:
We can't infer the type of a value injected into the generator. Although we trust redux-saga, it could be anything. That's why it won't infer the return value of the expression that gets evaluated by `yield`.

Reply:
Good point 👍